boards/posix/native_posix/sdl_model.c

```diff
@@ -0,0 +1,61 @@
+/* Copyright (c) 2018 dXplore
```

Comment:
Do you overall want to have this code as a HW model?
It feels to me it would more naturally fit as part of the driver.
If you need to run your initialization before Zephyr boots you can use `NATIVE_TASK` to register hwm_sdl_init to be called on `PREBOOT`.
You can create a Zephyr thread for the driver to update the window (or have a separate native pthread if you feel comfy enough)

Reply:
Problem is that if it is made part of the display driver it depends on the driver to be present.
It could be the case that the SDL event loop is used to simulate other HW behavior such as input events (e.g. user input via keyboard/mouse).

Creating a thread is definitely not an option as SDL expects that the events are polled from the main thread.

Reply:
(disclaimer I never played with SDL) In SDL can you not register a callback to be called when an event arrives instead of constantly polling for them?

> It could be the case that...

I guess one could imagine some other GUI based "HW models" for native_posix based on SDL. But as of now the only SDL based component is this proposed display driver. This split seems to be complicating things quite a bit. If possible I would propose the HAS_SDL code be made part of the driver by now, and if at some point in the future there is some other driver/model which also needs some of the SDL handling code, then that be moved to a common place.

Reply:
Yes there is callback support in SDL but to my surprise it still requires the poll loop (to be more precisely a periodic call to [SDL_PumpEvents](https://wiki.libsdl.org/SDL_PumpEvents)).

Actually this code came to live when I was looking into using SDL to capture keyboard events to interact with the GUI. As this code was already useful for this PR I just appended it to the commit list.

Reply:
So, a separate pthread (asynchronous to the other native_posix threads) dedicated to SDL, which would do the SDL initialization, and after, would have an infinite loop blocking on `SDL_WaitEvent()` would not be a possibility?

If you really need to have the polling code in the 1st thread, then, please do the timer thingy I mentioned above.

Reply:
Thx for all the input I will go for the timer thingy especially because of the warning in the SDL_PumpEvents API documentation: 
> WARNING: This should only be run in the thread that initialized the video subsystem, and for extra safety, you should consider only doing those things on the main thread in any case.

Reply:
@aescolar Tried a separate pthread and it seems to work